### DISCUSSION TITLE
Behavior of Annotated[Annotated[str, StringConstraints]]

### DISCUSSION THREAD

**Discussion by andersfylling** [pydantic/pydantic#7531] [Question] [answered]
Hello! I noticed that wrapping a Annotated constr will disable the validation. Is this desired behavior for pydantic? 
```py
def test_validation() -> None:
    class Faulty(BaseModel):
        v: Annotated[str, constr(max_length=2)]

    large_str = "123"
    Faulty(v=large_str)  # OK

    class Working(BaseModel):
        v: constr(max_length=2)

    Working(v=large_str)  # ERROR
```

**Comment by PHvL** [comment, marked as answer]
Shouldn't it be `Annotated[str, StringConstraints(max_length=2)]` (annotating type str with constraints as metadata) instead of `Annotated[str, constr(max_length=2)]` (annotating type str with a constrained str-type)

**Reply by samuelcolvin** [reply]
Ye, `constr` is designed to be used directly, it's now discouraged because it breaks static typing.
